cfme/containers/node.py

```diff
@@ -2,31 +2,102 @@
 # added new list_tbl definition
 from functools import partial
 from navmazing import NavigateToAttribute, NavigateToSibling
-
-from cfme.common import SummaryMixin, Taggable
+from widgetastic.widget import View, Text
+from widgetastic_manageiq import (
+    BootstrapSelect, Button, Table, Accordion, ManageIQTree, PaginationPane)
+from widgetastic_patternfly import Dropdown
+
+from cfme import BaseLoggedInPage
+from cfme.common import Taggable
+from cfme.containers.provider import ContainersProvider
+from cfme.exceptions import NodeNotFound
 from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import CheckboxTable, toolbar as tb, paginator, InfoBlock, match_location
+from cfme.web_ui import CheckboxTable, toolbar as tb, InfoBlock, match_location
 from utils.appliance import Navigatable
-from utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
-from cfme.containers.provider import pol_btn, mon_btn
+from utils.appliance.implementations.ui import CFMENavigateStep, navigator
+from utils.log import logger
 
 list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
 
 match_page = partial(match_location, controller='container_node', title='Nodes')
 
+# TODO Replace with resource table widget
+resource_locator = "//div[@id='records_div']/table//span[@title='{}']"
+
+
+class NodeView(BaseLoggedInPage):
+    title = Text('#explorer_title_text')
+
+    monitor = Dropdown('Monitoring')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
+
+    nodes = Table(locator="//div[@id='list_grid']//table")
+
+    @property
+    def in_cloud_instance(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Compute', 'Containers', 'Container Nodes'] and
+            match_page()  # No summary, just match controller and title
+        )
+
+
+class NodeCollection(Navigatable):
+    """Collection object for :py:class:`Node`."""
+
+    def instantiate(self, name, provider):
+        return Node(name=name, provider=provider, appliance=self.appliance)
+
+    def all(self):
```

Comment:
honestly, I imagined collections approach as a sigleton attached to appliance/provider, automatically updated and always providing me the same object if I ask for it.